Sources/MapboxMaps/Foundation/Extensions/Core/DownloadStatus.swift

```diff
@@ -0,0 +1,122 @@
+@_exported import MapboxCommon
+@_implementationOnly import MapboxCommon_Private
+
+extension DownloadError: LocalizedError {
+    /// Standardized error message
+    public var errorDescription: String? {
+        return "\(code): \(message)"
+    }
+}
+
+extension DownloadStatus {
+
+    /// Initialize a `DownloadStatus`
+    ///
+    /// - Parameters:
+    ///   - downloadId: Download id which was created by download request.
+    ///   - state: State of download request.
+    ///   - error: Contains error information in case of failure when state is
+    ///         set to `DownloadState.failed`.
+    ///   - totalBytes: Total amount of bytes to receive. In some cases this
+    ///         value is unknown until we get final part of the file.
+    ///   - receivedBytes: Amount of bytes already received and saved on the disk.
+    ///         Includes previous download attempts for a resumed download.
+    ///   - transferredBytes: Amount of bytes received during the current resume
+    ///         attempt. For downloads that weren't resumed, this value will be
+    ///         the same as receivedBytes.
+    ///   - downloadOptions: Download options used to send the download request.
+    ///   - httpResult: HTTP result. This field is only set for `DownloadState.failed`
+    ///         and `DownloadState.finished`.
+    ///         For `.failed` expect `HttpRequestError` to be provided for cases when
+    ///         `DownloadErrorCode` is `NetworkError`. For `.finished` `HttpResponseData`
+    ///         is set, but with empty data field (since all the data was written
+    ///         to the disk).
+    public convenience init(downloadId: UInt64,
+                            state: DownloadState,
+                            error: DownloadError?,
+                            totalBytes: UInt64?,
+                            receivedBytes: UInt64,
+                            transferredBytes: UInt64,
+                            downloadOptions: DownloadOptions,
+                            httpResult: Result<HttpResponseData, HttpRequestError>) {
+
+        let expected: Expected<AnyObject, AnyObject>
+        switch httpResult {
+        case let .success(response):
+            expected = Expected(value: response)
+        case let .failure(error):
+            expected = Expected(error: error)
+        }
+
+        self.init(downloadId: downloadId,
+                  state: state,
+                  error: error,
+                  totalBytes: totalBytes?.NSNumber,
+                  receivedBytes: receivedBytes,
+                  transferredBytes: transferredBytes,
+                  downloadOptions: downloadOptions,
+                  httpResult: expected)
+    }
+
+    /// Convenience to initialize a `DownloadStatus` when the download state is `.pending`
+    ///
+    /// - Parameters:
+    ///   - error: Contains error information in case of failure when state is
+    ///         set to `DownloadState.failed`.
```

Comment:
This is confusing. If this initializer is for use with download state pending, why would you ever provide an error if that's only for the failed state?